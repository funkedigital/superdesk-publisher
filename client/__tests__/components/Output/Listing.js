import React from "react";
import Listing from "../../../components/Output/Listing";
import { render, wait } from "@testing-library/react";
import Store from "../../../components/Output/Store";
import Publisher from "../../../__mocks__/publisher";

const publisher = new Publisher();

jest.mock("react-virtualized/styles.css", () => {
  return {};
});
jest.mock("axios");
jest.mock("react-select", () => props => "div");

let api = () => { };

api.users = {};
api.users.query = function () {
  return new Promise((resolve, reject) => {
    resolve({
      _items: [{ is_author: true, display_name: "test author" }],
      _links: {}
    });
  });
};

describe("Output/Listing", () => {
  it("renders properly", async () => {
    const { container, getByText } = render(
      <Store.Provider
        value={{
          publisher: publisher,
          api: api,
          tenants: [
            {
              name: 'tenant1', code: 'tenant1', routes: [
                { name: 'testroute1', id: 1 },
                { name: 'testroute2', id: 2 },
              ]
            },
            { name: 'tenant2', code: 'tenant2', routes: [] }
          ],
          filters: { sort: "updated_at", order: "desc" },
          actions: { setFilters: jest.fn(), setArticlesCounts: jest.fn() }
        }}
      >
        <Listing
          type="published"
          hide={false}
        />
      </Store.Provider>
    );

    await wait(() =>
      expect(container.querySelector(".sd-loader")).not.toBeInTheDocument(),
    )

    expect(container.firstChild).toMatchSnapshot();
  });
});
