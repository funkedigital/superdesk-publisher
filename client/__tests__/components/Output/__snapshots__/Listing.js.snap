// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output/Listing renders properly 1`] = `
<div
  class="sd-column-box__main-column relative"
  style="display: flex;"
>
  <div
    class="sd-list-item-group sd-shadow--z2"
    style="flex-grow: 1;"
  >
    <div
      style="overflow: visible; height: 0px; width: 0px;"
    >
      <div
        aria-label="grid"
        aria-readonly="true"
        class="ReactVirtualized__Grid ReactVirtualized__List"
        role="grid"
        style="box-sizing: border-box; direction: ltr; height: 0px; position: relative; width: 0px; will-change: transform; overflow-x: hidden; overflow-y: auto;"
        tabindex="0"
      />
    </div>
    <div
      class="resize-triggers"
    >
      <div
        class="expand-trigger"
      >
        <div
          style="width: 1px; height: 1px;"
        />
      </div>
      <div
        class="contract-trigger"
      />
    </div>
  </div>
</div>
`;
